module/zfs/zfs_byteswap.c

```diff
@@ -138,23 +136,17 @@ zfs_ace_byteswap(void *buf, size_t size, boolean_t zfs_layout)
 	}
 }
 
-/* ARGSUSED */
 void
 zfs_oldacl_byteswap(void *buf, size_t size)
 {
-	int cnt;
-
 	/*
 	 * Arggh, since we don't know how many ACEs are in
 	 * the array, we have to swap the entire block
 	 */
-
-	cnt = size / sizeof (ace_t);
-
-	zfs_oldace_byteswap((ace_t *)buf, cnt);
+	(void) buf;
+	zfs_oldace_byteswap((ace_t *)buf, size / sizeof (ace_t));
```

Comment:
`buf` is being used here.  The previous line `(void) buf` is wrong.  Since there may be other lines with this problem (it's hard to check), it would be good to be able to check them all.  But since a cast effectively disables all compiler warnings, the compiler can't help.  And since C casts are difficult to grep, a script will be difficult to write.  I suggest implementing and using a macro `UNUSED(x)`, which can later be easily grepped.

Reply:
Alas, this would've been maybe-sensible when I did the first round of this. I'm going to be doing a full re-review of all the diffs with a clear head anyway, I was in a bad state when I wrote the original (as can be seen from all the relatively-obvious issues you noted above that I completely missed then).

Reply:
You are still on time to use UNUSED.  You can create a patch set, apply sed to the patch set, which will be much easier than applying it to the whole code base, and then apply the patch set to the repository.

Something like:
```
sed -i '/^+/s/(void) \(\w\w*\\)\([,;]\)/UNUSED(\1)\2/g' 00*.patch
```

Reply:
This ship has, unfortunately, sailed with the first round of this in a previous PR (or two (or five?)), which I'm not redoing. It may've been a good idea to do that then, but if I do it for this now, I'd have to either (a) do a global void-cast pass replacing all of them (and I'd sooner die) or (b) I'll just forget and remove because it's another weird macro that's better-served with a void-cast.

Reply:
Fair enough.